src/sympc/store/crypto_store.py

```diff
@@ -0,0 +1,27 @@
+from typing import Any
+from typing import Callable
+from typing import Union
+from typing import List
+from typing import Tuple
+from typing import Dict
+
+from collections import defaultdict
+import itertools
+import operator
+
+
+class CryptoStore:
+    def __init__(self):
+        # The instances are kept in the same row
+        self.store: Dict[str, Any] = defaultdict(list)
+
+    def populate_store(self, key_store: str, primitives: Union[List[Any], Any]) -> None:
+        if isinstance(primitives, list):
+            self.store[key_store].extend(primitives)
+        else:
+            self.store[key_store].append(primitives)
+
+    def get_primitives_from_store(
+        self, key_store: str, nr_instances: int = 1
+    ) -> List[Any]:
+        return self.store[key_store][: nr_instances + 1]
```

Comment:
i'd be very careful with `nr_instances`, from our experience it's better to not slice at all (because it's highly inefficient).
For triples for example, you could had a list of primitives that you pop out when needed (like here: https://github.com/OpenMined/PySyft/blob/syft_0.2.x/syft/frameworks/torch/mpc/primitives.py#L82) 
So smthg like
```
matmul_primitive = {shape_type: [triple_0, triple_1, ....]} # you pop out `triple_0`
```
For comparison primitives, where you have nr_instances > 1, better would even be to add the nr_instances int the key:
```
comp_primitive = {<shape>_<type>_<nr_instances>: [prim_0, prim_1, ....]} # you pop out `prim_0`
```

Also, here in your code, you don't delete the primitive once it's used?

Reply:
Yep - I should also delete it after using it.

Do you have any idea if using ```numpy.views``` would provide a better runtime?

Reply:
not sure :/